src/main/java/org/broadinstitute/consent/http/resources/StatusResource.java

```diff
@@ -32,14 +37,37 @@ public Response getStatus() {
         Map<String, HealthCheck.Result> results = healthChecks.runHealthChecks();
         HealthCheck.Result postgresql = results.getOrDefault(DB_ENV, HealthCheck.Result.unhealthy("Unable to access postgresql database"));
         if (postgresql.isHealthy()) {
-            return Response.ok(results).build();
+            return Response.ok(formatResults(results)).build();
         } else {
             results.entrySet().
                     stream().
                     filter(e -> !e.getValue().isHealthy()).
                     forEach(e -> logger().error("Error in service " + e.getKey() + ": " + formatResultError(e.getValue())));
-            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(results).build();
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(formatResults(results)).build();
+        }
+    }
+
+    private Map<String, Object> formatResults(Map<String, HealthCheck.Result> results) {
+        // Order is important so we put Ok and Degraded states at the beginning of the map and
+        // add all other entries after that.
+        Map<String, Object> formattedResults = new LinkedHashMap<>();
+        boolean healthy = false;
+        HealthCheck.Result postgresql = results.getOrDefault(DB_ENV, HealthCheck.Result.unhealthy("Unable to access postgresql database"));
+        if (postgresql.isHealthy()) {
+            healthy = true;
         }
+        formattedResults.put(OK, healthy);
+        HealthCheck.Result gcs = results.getOrDefault(ConsentApplication.GCS_CHECK, HealthCheck.Result.unhealthy("Unable to access Sam"));
+        HealthCheck.Result elasticSearch = results.getOrDefault(ConsentApplication.ES_CHECK, HealthCheck.Result.unhealthy("Unable to access Sam"));
+        HealthCheck.Result ontology = results.getOrDefault(ConsentApplication.ONTOLOGY_CHECK, HealthCheck.Result.unhealthy("Unable to access Sam"));
+        HealthCheck.Result sam = results.getOrDefault(ConsentApplication.SAM_CHECK, HealthCheck.Result.unhealthy("Unable to access Sam"));
+        boolean degraded = (!gcs.isHealthy()
+                || !elasticSearch.isHealthy()
+                || !ontology.isHealthy()
+                || !sam.isHealthy());
+        formattedResults.put(DEGRADED, degraded);
+        formattedResults.putAll(results);
+        return formattedResults;
```

Comment:
Stacktraces are being included in the status message when some of the systems are unhealthy (in this case it's ElasticSearch). Not sure if we want external error logs showing up in our public status endpoint.
<img width="1781" alt="Screen Shot 2021-08-19 at 10 45 29 AM" src="https://user-images.githubusercontent.com/6664778/130090114-2a458ed8-e5ab-4fde-aeae-96074e6b9bb0.png">

Reply:
That's probably a good idea to clean that up. I think it might deserve its own ticket though. It should be easy enough to implement, it just means not passing the whole exception up to the status resource and generating a custom `Result.unhealthy()` response with a truncated exception message.